test_scripts/Defects/5_0/2347_HMI_sends_request_with_String_param_with_invalid_characters.lua

```diff
@@ -0,0 +1,86 @@
+---------------------------------------------------------------------------------------------------
+-- User story: https://github.com/SmartDeviceLink/sdl_core/issues/2347
+--
+-- Description:
+-- HMI sends request with String param with invalid characters.
+--
+-- Precondition:
+-- SDL and HMI are started.
+-- App is registered and activated.
+--
+-- In case:
+-- 1) HMI sends *request to SDL (please see list of impacted RPCs below) 
+-- and this request has at least one String param with '\n' and/or '\t' and/or 'whitespace' as the only symbol(s).
+--
+--  SDL.GetUserFriendlyMessage 
+--  SDL.GetDeviceConnectionStatus - NO STRING PARAMS (NOT VERIFIED)
+--  SDL.ActivateApp - NO STRING PARAMS (NOT VERIFIED)
+--  SDL.GetListOfPermissions - NO STRING PARAMS (NOT VERIFIED)
+--  SDL.UpdateSDL - NO STRING PARAMS (NOT VERIFIED)
+--  SDL.GetStatusUpdate - NO STRING PARAMS (NOT VERIFIED)
+--  SDL.GetURLs - NO STRING PARAMS (NOT VERIFIED)
+--
+-- Expected result:
+-- 1) Log corresponding error internally respond with 'INVALID_DATA' to HMI.
+--
+-- Actual result:
+-- N/A
+---------------------------------------------------------------------------------------------------
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local common = require('user_modules/sequences/actions')
+
+--[[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+--[[ Local Variables ]]
+local invalidParams = {
+	endline = {
+		language = "EN-US",
+		messageCodes = {"\n"}
+	},
+	tab = {
+		language = "EN-US",
+		messageCodes = {"\t"}
+	},
+	whitespace = {
+		language = "EN-US",
+		messageCodes = {" "}
+	}
+}
+
+--[[ Local Functions ]]
+local function testGetUserFriendlyMessageValidParams()
+	local rqIdValid = common.getHMIConnection():SendRequest("SDL.GetUserFriendlyMessage", {
+		language = "EN-US", 
+		messageCodes = {"DataConsent"}
+	})
+	common.getHMIConnection():ExpectResponse(rqIdValid, {result = {code = 0, method = "SDL.GetUserFriendlyMessage"}})
+end
+
+local function testGetUserFriendlyMessageInvalidMandatory(pMessageCodes)
+	local rqIdMandatory = common.getHMIConnection():SendRequest("SDL.GetUserFriendlyMessage", { messageCodes = pMessageCodes })
+	common.getHMIConnection():ExpectResponse(rqIdMandatory, {error = {code = 11, data = {method = "SDL.GetUserFriendlyMessage"}}})
+end
+
+local function testGetUserFriendlyMessageFullRequest(pRqParams)
+	local rqIdFull = common.getHMIConnection():SendRequest("SDL.GetUserFriendlyMessage", pRqParams)
+	common.getHMIConnection():ExpectResponse(rqIdFull, {error = {code = 11, data = {method = "SDL.GetUserFriendlyMessage"}}})
+end
+
```

Comment:
@KrupenichOleg I believe `testGetUserFriendlyMessageFullRequest()` and `testGetUserFriendlyMessageInvalidMandatory()` do the same job and could be combined into one.

Reply:
Done